example/client/burl/cookie.cpp

```diff
@@ -0,0 +1,346 @@
+//
+// Copyright (c) 2024 Mohammad Nejati
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/cppalliance/http_io
+//
+
+#include "cookie.hpp"
+
+#include <iomanip>
+#include <sstream>
+
+namespace{
+
+struct name_chars_t
+{
+    constexpr
+    bool
+    operator()(char c) const noexcept
+    {
+        return
+            c > 0x20 && c != 0x7F &&
+            c != '(' && c != ')'  && c != '<' && c != '>'  && c != '@' &&
+            c != ',' && c != ';'  && c != ':' && c != '\\' && c != '"' &&
+            c != '/' && c != '['  && c != ']' && c != '?'  && c != '=' &&
+            c != '{' && c != '}';
+    }
+};
+
+constexpr auto name_chars = name_chars_t{};
+
+struct value_chars_t
+{
+    constexpr
+    bool
+    operator()(char c) const noexcept
+    {
+        return
+            (c == 0x21             ) ||
+            (c >= 0x23 && c <= 0x2B) ||
+            (c >= 0x2D && c <= 0x3A) ||
+            (c >= 0x3C && c <= 0x5B) ||
+            (c >= 0x5D && c <= 0x7E);
+    }
+};
+
+constexpr auto value_chars = value_chars_t{};
+
+constexpr auto attr_chars =
+    urls::grammar::all_chars -
+    urls::grammar::lut_chars("\x1F\x7f;");
+
+bool
+domain_match(
+    core::string_view r_domain,
+    core::string_view c_domain,
+    bool subdomains) noexcept
+{
+    if(!subdomains)
+        return r_domain == c_domain;
+
+    if(c_domain.starts_with('.'))
+        c_domain.remove_prefix(1);
+
+    if(r_domain.ends_with(c_domain))
+    {
+        if(r_domain.size() == c_domain.size())
+            return true;
+
+        return r_domain[r_domain.size() - c_domain.size() - 1] == '.';
+    }
+
+    return false;
+}
+
+bool
+path_match(
+    core::string_view r_path,
+    core::string_view c_path) noexcept
+{
+    if(r_path.empty())
+        return true;
+
+    if(r_path.starts_with(c_path))
+    {
+        if(r_path.size() == c_path.size())
+            return true;
+
+        if(c_path.ends_with('/'))
+            return true;
+
+        return r_path[r_path.size() - c_path.size()] == '/';
+    }
+
+    return false;
+}
+
+} // namespace
+
+boost::system::result<cookie>
+parse_cookie(core::string_view sv)
+{
+    namespace ug = urls::grammar;
+
+    const auto parse_rs = ug::parse(
+        sv,
+        ug::tuple_rule(
```

Comment:
wouldn't it be better to calculate this value once and store it in a variable with static storage duration?